packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApiProxy.js

```diff
@@ -192,6 +185,63 @@ export class DhisInputSchemeResolvingApiProxy {
     return modifiedQuery;
   };
 
+  /**
+   * @param response {*}
+   * @param dataElementCodes {*}
+   * @returns {*}
+   * @private
+   */
+  translateDataElementIdsToCodesInResponse = async (response, dataElementCodes) => {
+    const dataElementIdToCode = {};
+
+    const dataElements = await this.models.dataSource.find({
+      code: dataElementCodes,
+      type: 'dataElement',
+    });
+
+    for (const dataElement of dataElements) {
+      if (dataElement.config.dhisId) {
+        dataElementIdToCode[dataElement.config.dhisId] = dataElement.code;
+      }
+    }
+
+    return translateElementKeysInEventAnalytics(response, dataElementIdToCode);
+  };
+
+  /**
+   * @param response {*}
+   * @returns {*}
+   * @private
+   */
+  translateOrgUnitIdsToCodesInResponse = async response => {
+    if (!response.rows.length) return response;
+    const newRows = [];
+
+    const orgUnitIdIndex = response.headers.findIndex(({ name }) => name === 'ou');
+    const orgUnitCodeIndex = response.headers.findIndex(({ name }) => name === 'oucode');
+    if (!orgUnitIdIndex || !orgUnitCodeIndex)
+      throw new Error("Can't read org unit id/code from dhis");
+
+    const dhisIds = response.rows.map(row => row[orgUnitIdIndex]);
+
+    const mappings = await this.models.dataServiceEntity.find({
+      [QUERY_CONJUNCTIONS.RAW]: {
+        sql: `config->>'dhis_id' in (${dhisIds.map(() => '?')})`,
+        parameters: dhisIds,
+      },
+    });
+
+    for (const row of response.rows) {
```

Comment:
Any reason you didn't use 
```suggestion
    const newRows = response.rows.map(row => {
```
?
I don't mind the `for/of`, but would prefer `newRows` to be defined just above it rather than at the top of the function

Reply:
Much better!